flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/BroadcastRecordWriterTest.java

```diff
@@ -21,30 +21,30 @@
 import org.apache.flink.core.io.IOReadableWritable;
```

Comment:
Is this commit fixing tests broken by the previous commit? Or is it refactor? If the first this commit be squashed with the previous one. If the latter could you add an better explanation what this commit is doing in the commit message?

Reply:
I guess you meant the third commit in this PR? I should explain the differences for these commits.

The second commit adds one whole test for `BroadcastRecordWriter`.

The third commit is just making the previous tests in `RecordWriterTest` work in the way of `BroadcastRecordWriter`. So it is a refactor work. Before refactor, all the tests would create `SelectorRecordWriter` instance when running in `BroadcastRecordWriterTest` . After refactor, we defined the `isBroadcastWriter` for creating `BroadcastRecordWriter` instance when running these tests in `BroadcastRecordWriterTest`. It could be squashed with the second commit. But I thought it would be clear for review separately.

The fourth commit is for simplifying the previous codes, actually it is not related to this PR. We could also submit a separate PR for it.

Reply:
I think I would squash the second with third commit. And submit a hotfix PR separately for the fourth commit. What do you think. The first commit should be solved in previous PR.

Reply:
> And submit a hotfix PR separately for the fourth commit. 

I'm fine with having hotfixes in one single PR

> The third commit is just making the previous tests in RecordWriterTest work in the way of BroadcastRecordWriter

So the `BroadcastRecordWriterTest` added in the 2nd commit, doesn't actually test the `BroadcastRecordWriter` until it's fixed in the 3rd commit? In that case definitely those two commits should be squashed.

Reply:
Not really, the new added test in `BroadcastRecordWriterTest` for 2nd commit actually tests the function for `BroadcastRecordWriter`. In this commit, the `BroadcastRecordWriterTest` also extends the `RecordWriterTest`, so it would run all the existing tests in `RecordWriterTest`, but the created writer is still the `SelectorRecordWriter`. 

In short words, `BroadcastRecordWriterTest` extends all the existing tests in `RecordWriterTest` and run them in the way of `SelectorRecordWriter`. So in the 3rd commit, I introduced the `isBroadcastWriter` in the constructor for creating the specific `RecordWriter` instance based on this tag. Then all the existing tests would run for `BroadcastRecordWriterTest` in the way of `BroadcastRecordWriter`.

Anyway, I think it still makes sense to squash these two commits. The separation is just for distinguishing the new added test and refactoring all the old tests.